crates/core/tedge_agent/src/agent.rs

```diff
@@ -352,9 +371,39 @@ impl Agent {
             let tedge_to_te_converter = create_tedge_to_te_converter(&mut mqtt_actor_builder)?;
             runtime.spawn(tedge_to_te_converter).await?;
 
-            let file_transfer_server_builder =
-                FileTransferServerBuilder::try_bind(self.config.http_config).await?;
-            runtime.spawn(file_transfer_server_builder).await?;
+            let state_dir = agent_state_dir(self.config.state_dir, self.config.config_dir);
+            //TODO: Migrate the existing `clean_start` setting which is C8Y specific without breaking backward compatibility.
+            let clean_start = true;
+            let main_device =
+                EntityRegistrationMessage::main_device(self.config.device_id.unwrap());
+            let entity_store = EntityStore::with_main_device_and_default_service_type(
+                mqtt_schema.clone(),
+                main_device,
+                self.config.service.ty.clone(),
+                state_dir,
+                clean_start,
+            )
+            .unwrap();
+            let entity_store_server = EntityStoreServer::new(entity_store);
+            let mut entity_store_actor_builder =
+                ServerActorBuilder::new(entity_store_server, &ServerConfig::default(), Sequential);
+            mqtt_actor_builder.connect_mapped_sink(
+                entity_manager::server::subscriptions(),
+                &entity_store_actor_builder,
+                |message| {
+                    EntityRegistrationMessage::new(&message).map(|reg_message| RequestEnvelope {
+                        request: EntityStoreRequest::Create(reg_message),
+                        reply_to: Box::new(NullSender),
+                    })
+                },
+            );
+
+            let file_transfer_server_builder = FileTransferServerBuilder::try_bind(
+                self.config.http_config,
+                &mut entity_store_actor_builder,
+                &mut mqtt_actor_builder,
+            )
```

Comment:
I'm surprised to see the `file_transfer_server` listening on both HTTP and MQTT.

I would expect the `entity_store_actor` to listen to MQTT plus a channel of requests forwarded by the `file_transfer_server`. 

- What has been your drivers for this design?
- Is this because on the HTTP side a response is expected?

Reply:
> What has been your drivers for this design?

It might be just a naming confusing during this transition period. Even though the entire HTTP code is currently kept under the module named `file_transfer_server`, the plan is to rename this module into `http_server` and associated structs into `HttpServerBuilder`,`HttpServerActor`, etc. Then this `HttpActor` would delegate HTTP requests to `file_transfer_service` and `entity_store_service` respectively. When that happens, the `file_transfer_service` will no have any MQTT aspects in it.

Reply:
I understand now where we are not aligned:

- The responsibility to publish over MQTT entity state should be moved  from the `entity_store_router` HTTP handler to the `EntityStoreServer` actor.
- See https://github.com/thin-edge/thin-edge.io/pull/3230#discussion_r1900976673

```suggestion
            let file_transfer_server_builder = FileTransferServerBuilder::try_bind(
                self.config.http_config,
                &mut entity_store_actor_builder,
            )
```

Reply:
Resolved by 1a85933bd5b546dc611ac9c93230493ad2b6a284

Reply:
Thanks. This simplifies a lot the file transfer server code.